### PR TITLE
Setup e2e test with cypress

### PR DESCRIPTION
**Pull request by MonPote**
Nothing to test, since all the work is done in the CI.
However a review of the code is appreciated.
Please note that dependency is fetch in `ui/cypress.sh ` and not in `requirement.sh` because it breaks  tests.

### CODE REVIEW

**Comment by MonPote** [comment]
@bert-e reset

**Comment by gdemonet** [comment]
> dependency is fetch in ui/cypress.sh and not in requirement.sh because it breaks tests

Mind elaborating?

File: ui/cypress.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# provides libasound
+yum install -y alsa-lib*
+yum install -y epel-release
+
+yum install -y gcc-c++ make
+curl -sL https://rpm.nodesource.com/setup_10.x | bash -
+yum install -y nodejs
+
+# Remove the package.json to avoid installing useless dependency
+rm package.json package-lock.json
```
Comment (gdemonet) [inline ui/cypress.sh]: Mmmh not really fond of this, it really is a hack. Did you try something like this?

```
npm install --no-save --quiet --no-package-lock <your_pkg>
```

Reference: https://stackoverflow.com/questions/52786695/npm-install-always-installs-everything-from-package-json

File: ui/cypress/integration/login.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+describe('Login test', function () {
+  it('should login', function () {
+    let target_url = Cypress.env('target_url');
+    cy.visit(target_url);
+
+    // FIXME Need to find a workaround for the api certification
```
Comment (gdemonet) [inline ui/cypress/integration/login.js:6]: Do you have a tracking ticket?

File: ui/package.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
     "@kubernetes/client-node": "github:scality/kubernetes-client-javascript.git#browser-dist",
     "axios": "^0.18.0",
     "core-ui": "github:scality/core-ui.git",
+    "cypress": "^3.2.0",
```
Comment (gdemonet) [inline ui/package.json]: No, that should be in `devDependencies`

File: ui/cypress/integration/login.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+describe('Login test', function () {
+  it('should login', function () {
+    let target_url = Cypress.env('target_url');
+    cy.visit(target_url);
```
Comment (gdemonet) [inline ui/cypress/integration/login.js:4]: So, visiting works... Don't you need to set-up anchors for filling the login form? Will it be done in a later ticket?

**Review by gdemonet — CHANGES_REQUESTED** [review]
A few changes are required, also not really fond of the `ui/cypress.sh` installing everything. Running the `npm install cypress` should be enough IMHO.

**Comment by MonPote** [comment]
@bert-e reset

**Comment by MonPote** [comment]
@bert-e reset

File: ui/cypress.sh (hunk 1 above)
Comment (MonPote) [reply ui/cypress.sh]: Thanks for the tips

File: ui/package.json (hunk 3 above)
Comment (MonPote) [reply ui/package.json]: 👍

File: ui/cypress/integration/login.js (hunk 4 above)
Comment (MonPote) [reply ui/cypress/integration/login.js:4]: Yep it will be done in this ticket #1107

File: ui/cypress/integration/login.js (hunk 2 above)
Comment (MonPote) [reply ui/cypress/integration/login.js:6]: #1107

**Comment by MonPote** [comment]
@bert-e reset

**Comment by MonPote** [comment]
@bert-e approve

**Review by gdemonet — APPROVED** [review]
LGTM!

**Pull request merged**